### PR TITLE
Add stress.reporter — pure stats for the upcoming stress harness

### PR DESCRIPTION
**Pull request by julietshen**
## Summary

Pure-function stats library that takes per-event produce/consume timestamps and emits a structured report: drop count and rate, percentile latencies, threshold-breach detection, JSON + human format, and an exit-code policy for CI gating.

Designed to be producer-agnostic so the same module serves both:
* **closed-loop** measurement (known set of action_ids, exact 1:1 matching)
* **open-loop** measurement (count anything that shows up in a time window)

No Kafka, no I/O, no live deps. Fully unit-tested in isolation.

Part 2 of 5 in the stress-harness split for #324, per AGENTS.md (\"Keep diffs small and focused\"). Other parts:
- [#327](https://github.com/roostorg/osprey/pull/327): `GetActionId` UDF
- (this PR): reporter
- (to come): producer
- (to come): consumer
- (to come): CLI + entry point

This PR is independent — does not depend on #327 or any sibling PR. The next PRs (producer, consumer, CLI) will import from this module.

## Test plan

- [x] 15 unit tests covering: percentile math edge cases, closed-loop vs open-loop discrimination, drop-rate calculation, threshold breach detection, JSON roundtrip, human format
- [x] `uv run ruff check` — clean
- [x] `uv run mypy` — clean
- [ ] CI runs the full test suite

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added a stress-harness reporter that calculates match/drop metrics from event timestamps and computes latency distributions (min, p50, p95, p99, max).
  * Supports optional thresholds for drop rate and p95 latency, returning deterministic CLI exit codes when limits are exceeded.
  * Generates both JSON output and a human-readable summary, including closed-loop vs open-loop reporting behavior.
* **Tests**
  * Added unit tests covering percentile validation, latency aggregation, reporting modes, threshold breach detection and exit codes, and JSON/human serialization.

### CODE REVIEW

**Comment by julietshen** [comment]
Addressed CodeRabbit's two notes in `76b31a1`:

**1. `compute_latency_stats` validation (accepted).** Added an explicit `ValueError` for empty input, matching `percentile()`'s existing pattern, plus a unit test. Agreed the public surface deserves a consistent error rather than letting `IndexError` leak.

**2. Pydantic models for `LatencyStats` / `Thresholds` / `Report` (declining).** These are internal pure-data containers — no validation, no parsing from untrusted input, no API serialization beyond `json.dumps(asdict(...))`. The codebase uses pydantic where validation/serialization at boundaries matters (input schemas, persisted models), and uses `@dataclass` elsewhere — e.g. `ExecutionResult`, `Action`, `LatencyStats`'s shape neighbors in `execution_context.py`. Switching to `pydantic.BaseModel` here would add inheritance overhead, validator machinery, and v1's known performance cost without anything to validate. If we hit a real boundary (e.g. a future REST endpoint that returns `Report`), we can wrap it in a pydantic model at the boundary; the internal type doesn't need to change.

Lint, mypy, and the 16 reporter tests still pass.

File: osprey_worker/src/osprey/worker/stress/tests/test_reporter.py [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+        stats = compute_latency_stats([100.0, 50.0, 25.0, 75.0, 200.0])
+        assert stats.min_ms == 25.0
+        assert stats.max_ms == 200.0
+        assert stats.count == 5
+        # Sorted: 25, 50, 75, 100, 200; p50 = sorted[ceil(2.5)-1] = sorted[2] = 75
+        assert stats.p50_ms == 75.0
+
+    def test_raises_on_empty(self) -> None:
+        # Match percentile()'s ValueError instead of letting an IndexError leak;
+        # this function is public and callers deserve a clear error.
```
Comment (chimosky) [inline osprey_worker/src/osprey/worker/stress/tests/test_reporter.py]: This seems redundant, the code does show what it does and I don't see the point of the comment.

**Comment by chimosky** [comment]
Reviewed 76b31a1.

File: osprey_worker/src/osprey/worker/stress/reporter.py [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+    """Nearest-rank percentile over a pre-sorted list. p in [0, 100]."""
+    if not sorted_values:
+        raise ValueError('percentile of empty list')
+    if not 0 <= p <= 100:
+        raise ValueError(f'p must be in [0, 100], got {p}')
+    # Nearest-rank: ceil(p/100 * n) - 1, clamped.
+    n = len(sorted_values)
+    if p == 0:
+        return sorted_values[0]
+    rank = int(-(-p * n // 100)) - 1  # ceil division
```
Comment (reitblatt) [inline osprey_worker/src/osprey/worker/stress/reporter.py:91]: TIL: good trick for ceil divison in python!

**Comment by julietshen** [comment]
Rebased on latest main and addressed @chimosky's note in `05f5c3b` — dropped the redundant comment block above `test_raises_on_empty`. The test name + assertion already convey the contract.

16/16 reporter tests still pass; ruff + mypy clean.

(@reitblatt - i learned something while working on this too! 🙂)

**Review by cassidyjames — APPROVED** [review]
:shipit:
